Fix bugs with prompter

bjoginapally opened the pull request:
# Description
- Fix filter feature on list prompts
- Print the original prompt for a summary view of selected values
- Enable back option while selecting cloud provider
- Fix default option for text inputs
- Remove "default" kubecontext item from list

## Issue reference

<!--
We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.
-->

Fixes: NA

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Adds necessary unit tests for change
* [ ] Adds necessary E2E tests for change
* [x] Unit tests passing
* [ ] Extended the documentation / Created issue for it

Commit:
Fix bugs with prompter
  - Fix filter feature on list prompts
  - Print the original prompt for a summary view of selected values
  - Enable back option while selecting cloud provider
  - Fix default option for text inputs
  - Remove "default" kubecontext item from list

issue: bugs

rynowak reviewed (dismissed):
Where are the test changes? If these are functional bugs we should add unit tests so they don't regress again.

Commit:
Correct bubble tea exit behavior
- Bubble tea now exits init command when quit input is entered
- Added unit tests

Commit:
Remove double : in some prompts and make class constants as private

Commit:
Misc changes

ytimocin reviewed (approved):
lgtm but needs a rebase

Commit:
Change env name prompt function to be generic
